HAPNodeJSClient.js

```diff
@@ -294,40 +302,60 @@ function _reconnectServer(server) {
 
-  var reconnectTimer;
   if (server.deviceID) {
-    reconnectTimer = setInterval(function() {
-      this.HAPeventByDeviceID(server.deviceID, JSON.stringify({
+    if (reconnectTimer.byDevice.hasOwnProperty(server.deviceId)) {
+      clearInterval(reconnectTimer.byDevice[server.deviceId]);
+    }
+
+    reconnectTimer.byDevice[server.deviceId] = setInterval(function() {
+      this.HAPeventByDeviceID(server.deviceID, pinCache.byDevice[server.deviceID], JSON.stringify({
         characteristics: this.eventRegistry[server.deviceID]
-      }), clearTimer);
+      }), function(err, rsp) {
+        if (err) {
+          debug('HAPevent event reregister failed, retry in 60', server);
+        } else {
+          debug('HAPevent event reregister succeeded', server);
+          clearInterval(reconnectTimer.byDevice[server.deviceId]);
+          delete reconnectTimer.byDevice[server.deviceId];
+        }
+      });
     }.bind(this), 60000);
   } else {
-    reconnectTimer = setInterval(function() {
-      this.HAPevent(server.server.split(':')[0], server.server.split(':')[1], JSON.stringify({
-        characteristics: this.eventRegistry[server.server]
-      }), clearTimer);
-    }.bind(this), 60000);
-  }
+    var ipAddress = server.server.split(':')[0];
+    var port = server.server.split(':')[1];
+    var key = ipAddress + ':' + port;
 
-  function clearTimer(err, rsp) {
-    if (err) {
-      debug('HAPevent event reregister failed, retry in 60', server);
-    } else {
-      debug('HAPevent event reregister succeeded', server);
-      clearInterval(reconnectTimer);
+    if (reconnectTimer.byIp.hasOwnProperty(key)) {
+      clearInterval(reconnectTimer.byIp[key]);
     }
+
+    reconnectTimer.byIp[key] = setInterval(function() {
+      this.HAPevent(ipAddress, port, pinCache.byIp[key], JSON.stringify({
+        characteristics: this.eventRegistry[server.server]
+      }), function(err, rsp) {
+        if (err) {
+          debug('HAPevent event reregister failed, retry in 60', server);
+        } else {
+          debug('HAPevent event reregister succeeded', server);
+          clearInterval(reconnectTimer.byIp[key]);
+          delete reconnectTimer.byIp[key];
+        }
+      });
+    }.bind(this), 60000);
   }
 }
 
 /**
  * HAPNodeJSClient.prototype.HAPeventByDeviceID - Send a characteristic PUT Message to a particular homebridge instance, this maintains a socket connection for use in returning Events
  *
  * @param  {type} deviceID  deviceID homebridge instance
+ * @param  {type} pin       Pin of homebridge instance
  * @param  {type} body      An array of HomeKit characteristic updates, [{ \"aid\": 2, \"iid\": 9, \"value\": 0}]
  * @param  {type} callback  Callback to execute upon completion of characteristic setting, function(err, response)
  */
 
-HAPNodeJSClient.prototype.HAPeventByDeviceID = function(deviceID, body, callback) {
+HAPNodeJSClient.prototype.HAPeventByDeviceID = function(deviceID, pin, body, callback) {
```

Comment:
This is a breaking change to the external API and can not be merged as it will break all other packages using this package